src/WebUI/Common/KubeSummary.tsx

```diff
@@ -273,11 +273,65 @@ export class KubeSummary extends BaseComponent<IKubeSummaryProps, IKubernetesCon
 
     private _onSelectionStoreChanged = () => {
         const selectionStoreState = this._selectionStore.getState();
-        this.setState({
-            showSelectedItem: selectionStoreState.showSelectedItem,
-            selectedItem: selectionStoreState.selectedItem,
-            selectedItemType: selectionStoreState.selectedItemType
-        });
+
+        // If the selection only has id and doesn't have the item, we would need to fetch the item
+        if (!!selectionStoreState.itemUID && !selectionStoreState.selectedItem) {
+            let invokedFunc: (() => Promise<V1DaemonSetList | V1ReplicaSetList | V1StatefulSetList>) | undefined = undefined;
+            let item: V1DaemonSet | V1ReplicaSet | V1StatefulSet | undefined = undefined;
+            switch (selectionStoreState.selectedItemType) {
+                case SelectedItemKeys.ReplicaSetKey:
+                    const replicaSets = this._workloadsStore.getState().replicaSetList;
+                    if (replicaSets && replicaSets.items && replicaSets.items.length) {
+                        item = replicaSets.items.filter(r => r.metadata.uid === selectionStoreState.itemUID)[0];
```

Comment:
is find() a better option here? as filter will go through all of the items. same with all other items. also this is copy of the code, do we have an anonymous function?